src/modules/checkIsValidUserInput.js

```diff
@@ -9,7 +9,31 @@
  * @return {boolean} - True if the user input is valid, false otherwise
  */
 function checkIsValidUserInput(userInput) {
-  /* Write your code here */
+  if (isNaN(+userInput)) {
+    return false;
+  }
+
+  if (userInput.length !== 4) {
+    return false;
+  }
+
+  if (userInput[0] === '0') {
+    return false;
+  }
+
+  for (let i = 0; i < userInput.length; i++) {
+    for (let j = 0; j < userInput.length; j++) {
+      if (i === j) {
+        continue;
+      }
+
+      if (userInput[i] === userInput[j]) {
+        return false;
+      }
+    }
```

Comment:
The nested loop used to check for duplicate digits can be optimized. Currently, it checks each pair of digits twice, which is unnecessary. Consider using a Set to track seen digits, which can reduce the complexity and improve performance.